New Traitor Items: Sith Starter Kit and X-TREME Laser Tag Kits

Surrealaser opened the pull request:
:cl:
add: New Traitor Items added: Sith Starter Kit exclusive to the Chaplain and Curator and X-TREME Laser Tag Kits exclusive to the Clown, Mime and Assistant.
add: Sith Starter Kit grants telekinesis, an improved lightsaber, and the force lightning ability for 19 TC.
tweak: Force Lightning inflicts stamina damage on the user. It can be deflected or reflected with an energy sword.
add: X-TREME Laser Tag kit that has energy weapon reflective vest and helmet and laser tag gun that has a lethal and stun mode, as well as a parental lock toggled by a swiped ID that (un)locks the fire selection. While locked it's set to its harmless laser tag mode. The first swipe imprints the ID that can activate this toggle. This ID can be reset with a multitool while it's unlocked, or an emag at any time.
add: There is a bulk Party Pack version of the X-TREME Laser Tag kit for 50 TC during Clown Ops.
/:cl:

Surrealaser committed:
New Traitor Items: Sith Starter Kit and X-TREME Laser Tag Kits

:cl:
add: New Traitor Items added: Sith Starter Kit exclusive to the Chaplain and Curator and X-TREME Laser Tag Kits exclusive to the Clown, Mime and Assistant.
add: Sith Starter Kit grants telekinesis, an improved lightsaber, and the force lightning ability for 20 TC.
add: X-TREME Laser Tag kit grants a laser tag gun that has a lethal and stun mode, as well as a parental lock activated by a swiped ID that locks the fire selection and sets it to its harmless mode.
add: There is a bulk Party Pack version of the X-TREME Laser Tag kit for 50 TC during Clown Ops.
/:cl:

Surrealaser committed:
Added Emag and Multitool functionality

ATH1909 commented:
https://github.com/tgstation/tgstation/blob/master/code/game/objects/items/storage/uplink_kits.dm

```
		if("darklord")
			new /obj/item/twohanded/dualsaber(src)
			new /obj/item/dnainjector/telemut/darkbundle(src)
			new /obj/item/clothing/suit/hooded/chaplain_hoodie(src)
			new /obj/item/card/id/syndicate(src)
			new /obj/item/clothing/shoes/chameleon/noslip(src) //because slipping while being a dark lord sucks
			new /obj/item/book/granter/spell/summonitem(src)
```

*cough* Syndi-Kit B *cough*

Admittedly, it's nice to be able to play a sith lord as a traitor without having to roll the right crate for your Syndi-Kit, but this is at least something worth knowing the existence of.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/deadly_laser_tag.dm:
```diff
@@ -0,0 +1,274 @@
+
+	if(!ID_imprint)
+		to_chat(user, "<span class='notice'>Parental lock ID imprinted. Swipe again with imprinted ID to toggle the parental lock feature.</span>")
+		ID_imprint = I
```
shouldn't this also set modifystate?

Surrealaser replied:
No. Modify state is only modified when the parental lock is specifically activated or deactivated.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/deadly_laser_tag.dm:
```diff
@@ -0,0 +1,274 @@
+	new /obj/item/gun/energy/laser/redtag/deadly(src)
+
+/obj/item/storage/box/syndicate/laser_tag_kit_blue/PopulateContents()
+	new /obj/item/clothing/suit/bluetag/deadly_laser_tag(src)
```
no helmet(s)?

Surrealaser replied:
Ah, forgot about those, will add.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+		to_chat(user, "<span class='warning'>No target found in range!</span>")
+		return
+	spark_setup()
+	user.say(pick("My hate has made me powerful...","I AM THE SENATE!!", "If you will not be turned... you will be destroyed!!", "You will pay the price for your lack of vision!!", "Young fool, only now, at the end, do you understand.", "Your feeble skills are no match for the power of the Dark Side!", "UNNNNLIMITEDDDD POWAHHHH!!", "And now young Skywalker, you will die."))
```
shouldn't there be a comma after "now"?

Surrealaser replied:
For which comment? Now young Skywalker?

ATH1909 replied:
Yeah, that one.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+	if(!can_cast(user))
+		return
+
+	if(iscarbon(user))
```
check to see if they have at least one hand to shoot the lightning from

Surrealaser replied:
Sure.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+			//to_chat(world, "Target [M] added.") //DEBUG
+	if(!possible_targets.len)
+		return
+	var/mob/living/next = pick(possible_targets)
```
can this rebound back to the caster?

Surrealaser replied:
No. I will consider changing this; probably when I add the block chance for light saber. I passed on it thus far because it's very niche.

ATH1909 replied:
No, I mean, can this (unintentionally) choose the CASTER as the second target of the spell/in the chain?

Surrealaser replied:
No. Impossible.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+
+
+///Triggers on COMSIG_MOB_ATTACK_RANGED. Usually handles stuff like picking up items at range.
+/datum/martial_art/starter_sith/proc/on_ranged_attack(datum/source, atom/target)
```
instead of making this be a part of this weird, unneeded martial art, just give them a TK injector (or give them the TK mutation when they finish reading the force lightning book).

Surrealaser replied:
As discussed on discord, because of the way the TK mutation is integrated into the code, the martial art is the best way I've found to implement this property, especially since it can also feature other properties.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+
+	to_chat(usr, "<b><i>You retreat inward and recall the teachings of the Sith...</i></b>")
+
+	to_chat(usr, "<span class='notice'>Force Lightning</span>: Fires a bolt of force lightning at a random target within 3 tiles.")
```
What if they read the second book but not the first?

Surrealaser replied:
I'll make a note about the Force Lightning sacred text being sold separately in the description.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+	. = ..()
+	UnregisterSignal(H, COMSIG_MOB_ATTACK_RANGED)
+
+/obj/item/melee/transforming/energy/sword/saber/sith
```
Just use a regular red esword, jeeze. Why can muggles deflect projectiles with normal eswords, but not a sith one? There's no reason to make a special snowflake sword to make SURE that a muggle will never be able to deflect projectiles with THIS SPECIFIC KIND OF ESWORD, especially since they can still just make normal melee attacks with it.

Plus, having your own weapons/powers able to be turned against you is totally in-theme for a Palpatine reference (see: how Palpatine was defeated three times in the movies).

Surrealaser replied:
Blocking works as normal with  the Sith sword; the one difference is that someone with Sith training has a 30% chance to _reflect_ projectiles using the Sith sword. There are also other qualities the Sith sword has that requires a subtype.

ATH1909 commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,250 @@
+
+	playsound(get_turf(current), 'sound/magic/lightningshock.ogg', 50, TRUE, -1)
+
+	if(current.anti_magic_check())
```
add a reference for if you try to use force lightning on someone who's holding an esword.

Surrealaser replied:
I'd probably give it a % chance to block equal to the esword's block chance.

ATH1909 commented on code/modules/uplink/uplink_items_fulp.dm:
```diff
@@ -30,7 +30,56 @@
+	cost = 20
+	restricted_roles = list("Chaplain", "Curator")
+
+/datum/uplink_item/role_restricted/laser_tag_kit_red
```
Suggestion: Make this a purchasable cargo crate instead of a traitor item. You'd want to nerf the extreme laser tag vests a bit if you do that, of course, but "toys that can be used as lethal weapons" sounds really on-theme for Cargo.

Surrealaser replied:
No; Cargo shouldn't have access to this. It's meant to be a semi-stealthy traitor item.

ATH1909 commented:
That entire second book (and that sith teachings "martial art") doesn't need to exist; its abilities can be moved into other items/effects.

Surrealaser commented:
> That entire second book (and that sith teachings "martial art") doesn't need to exist; its abilities can be moved into other items/effects.

See comments regarding TK.

gonenoculer5 commented:
@Surrealaser Hey surreal, I was actually going to package something similar to the curator thing into a compact item called a sith holocron, and add it as a part of my role-restricted updates lmao, if youd like we can sort of stuff that shit into my update and release the PR as a whole? Up to you

Surrealaser committed:
Updated descriptions and mechanics; added laser helm

Force lightning can now be blocked/deflected with a light saber.

XTREME laser kit now has appropriate helmets.

Surrealaser committed:
Added support for bonuses from force training

Force training now provides a bonus chance to deflect/reflect force lightning.

Surrealaser commented:
@gonenoculer5 Probably better to keep them separated for now, unless there's a compelling reason to merge them.

remove32 commented:
something i noticed that concerned me was the fact that the lightning has almost no cooldown. this is kinda offset by the low damage and the stamina drain on yourself for using it repeatedly.... still almost no cooldown makes it very easy to spam this

Surrealaser commented:
@remove32 Yep, that's by design. It's meant to feel like real force lightning, weak but spammable. The stamina damage that results in slowdown and being easier to stun is meant to help compensate for this.

Surrealaser committed:
More force lightning RNG exclamations

Surrealaser committed:
More!

SkippuSetsuna commented on code/FulpstationCode/Fulp_traitor_items/traitor_chaplain.dm:
```diff
@@ -0,0 +1,285 @@
+	spark_system.start()
+	if(isliving(user))
+		var/mob/living/L = user
+		L.electrocute_act(5,"Force Lightning",1,SHOCK_NOSTUN) //Snappy zappy.
```
After playing around with it I feel the cooldown for the spell is too quick and people can end up spamming it.

Surrealaser replied:
See above.

remove32 commented:
gameplay impacting - requesting review from swain

Surrealaser committed:
Conflict resolution

Surrealaser commented:
@remove32 Conflicts resolved

remove32 commented:
@TheSwain new antag gear - we should get you to try it in the test server

TheSwain commented:
Okay, let's give this one a go on the main server and see what happens. @Surrealaser as soon as the conflict gets resolved, i'll merge

Surrealaser committed:
Conflict Resolution

Surrealaser commented:
@TheSwain Conflict resolved. I think I'll decrement the TC cost by 1 point if only so the buyer can get one utility item.

Surrealaser committed:
TC and Description Tweak
